### ISSUE TITLE
Allow to use OIDC Client Filter and OIDC Token Propagation extensions in the same app

### ISSUE THREAD

**Issue by Sgitario** [quarkusio/quarkus#14466]
**Description**
At the moment, if we add these both extensions in the same application:

```java
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-oidc-client-filter</artifactId>
</dependency>
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-oidc-token-propagation</artifactId>
</dependency>
```

And we create a couple of RestClients for each:

```java
@RegisterRestClient
@OidcClientFilter
public interface AutoAcquireTokenPongClient {
    @GET
    @Path("/rest-pong")
    @Produces(MediaType.TEXT_PLAIN)
    String getPong();
}
```

```java
@RegisterRestClient
@AccessToken
public interface TokenPropagationPongClient {
    @GET
    @Path("/rest-pong")
    @Produces(MediaType.TEXT_PLAIN)
    String getPong();
}
```

The application fails to start with:

```
java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Multiple producers of item class io.quarkus.restclient.deployment.RestClientAnnotationProviderBuildItem (io.quarkus.oidc.client.filter.deployment.OidcClientFilterBuildStep#registerProvider)
Caused by: java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Multiple producers of item class io.quarkus.restclient.deployment.RestClientAnnotationProviderBuildItem (io.quarkus.oidc.client.filter.deployment.OidcClientFilterBuildStep#registerProvider)
Caused by: io.quarkus.builder.ChainBuildException: Multiple producers of item class io.quarkus.restclient.deployment.RestClientAnnotationProviderBuildItem (io.quarkus.oidc.client.filter.deployment.OidcClientFilterBuildStep#registerProvider)
Caused by: java.lang.Throwable: This is the location of the conflicting producer (io.quarkus.oidc.token.propagation.deployment.OidcTokenPropagationBuildStep#registerProvider)
```

I didn't report this as a bug as it's more like a Rest Client extension limitation, more than a bug.
Reproducer in: https://github.com/Sgitario/beefy-scenarios/tree/master/013-quarkus-oidc-restclient

**Implementation ideas**
The provider is based on the annotation present at the custom RestClient implementation, so it should be possible to refactor either the RestClient extension in order to accept multiple producers (not sure if this can be done) or create a composed producer somehow.

**Comment by sberyozkin** [comment]
Hi @Sgitario Yeah, what came with the OIDC client only allows for one annotation/producer at a time, if you have one client with 2 annotations or as in your case 2 clients, each of them with its own annotation, then it won't work.
I've asked for a first PR be opened for #14141...
thanks

**Comment by sberyozkin** [comment]
@Sgitario also, this fix, #14371, will need to be backported to 1.11.1 :-). For now please check against the snapshots, if Guillaume agrees then it will be backported and then you'd check it with 1.11.1
thanks

**Comment by Sgitario** [comment]
@sberyozkin thanks!
